hack/save-logs-files.sh

```diff
@@ -0,0 +1,236 @@
+#!/bin/bash
+
+# Copyright 2023 IBM Corp.
+# SPDX-License-Identifier: Apache-2.0
+
+# error and exit configuration
+set -eu
+
+FLAG_DURATION="d"
+FLAG_NAMESPACES="n"
+FLAG_PATH="p"
+FLAG_APP="a"
+FLAG_UUID="u"
+FLAG_HELP="h"
+ARGS_SEPARATOR=","
+APP_CONFLICT_MESSAGE="Only one of -$FLAG_APP/-$FLAG_UUID may be used"
+
+# set initial values of the flags
+duration=1h
+namespaces_list=()
+output_path=$(readlink -f .)/
+fybrik_application_ns=""
+fybrik_application_name=""
+fybrik_application_uuid=""
+
+function print_description {
+    echo -e "A script for collecting relevant fybrik logs and information into a local directory, using kubectl.\n"
+}
+
+function print_usage {
+    echo -e "Usage:\t\t$(basename $0) [-$FLAG_DURATION <duration>] [-$FLAG_NAMESPACES <ns1,ns2,...>] [-$FLAG_PATH <path>] [-$FLAG_APP <ns,app-name> OR -$FLAG_UUID <app-uuid>]";
+    echo -e "Default:\t$(basename $0) -$FLAG_DURATION 1h -$FLAG_NAMESPACES <any namespace with the word 'fybrik'> -$FLAG_PATH ./";
+}
+
+function print_flags {
+    echo -e "Flags:"
+    echo -e "\t-$FLAG_DURATION\t the relative duration of time (e.g. 5s, 2m, 3h), only logs newer than the duration will be saved in the output. Default: 1h"
+    echo -e "\t-$FLAG_NAMESPACES\t comma-separated namespaces, only resources from those namespaces will be saved in the output. Default: all the namespaces that contains the word 'fybrik'"
+    echo -e "\t-$FLAG_PATH\t a path of an existant directory in which the output directory will be created. Default: current working directory"
+    echo -e "\t-$FLAG_APP\t comma-separated namespace and name of fybrikapplication, for saving its yaml if the resource exists, and creating an additional file of the logs containing its uuid. Default: none. $APP_CONFLICT_MESSAGE"
+    echo -e "\t-$FLAG_UUID\t an app.fybrik.io/app-uuid for saving its yaml if the resource exists in the namespaces, and creating an additional file of the logs containing the uuid. Default: none. $APP_CONFLICT_MESSAGE"
+    echo -e "\t-$FLAG_HELP\t print this message and exit"
+}
+
+while getopts "$FLAG_DURATION:$FLAG_NAMESPACES:$FLAG_PATH:$FLAG_UUID:$FLAG_APP:$FLAG_HELP" opt; do
+    case "$opt" in
+        $FLAG_DURATION)
+            duration="$OPTARG"
+            # check that the argument is valid
+            re='^[0-9]+[smh]$'
+            if [[ ! $duration =~ $re ]]; then
+                echo "error: invalid argument '$duration' for $FLAG_DURATION flag: should contain a number and a time unit (s/m/h), e.g. 1h or 30m"
+                print_usage
+                exit 1
+            fi
+            ;;
+        $FLAG_NAMESPACES)
+            readarray -d $ARGS_SEPARATOR -t namespaces_list < <(printf '%s' "$OPTARG")
```

Comment:
./save-logs-files.sh: line 58: readarray: command not found

Reply:
Here are several possible solutions: https://stackoverflow.com/questions/23842261/alternative-to-readarray-because-it-does-not-work-on-mac-os-x